lib/Wikia/src/Service/User/Preferences/PreferenceModule.php

```diff
@@ -18,7 +18,19 @@ public function configure(InjectorBuilder $builder) {
 				return $wgHiddenPrefs;
 			})
 			->bind(UserPreferences::DEFAULT_PREFERENCES)->to(function() {
-				return User::getDefaultOptions();
+				global $wgUserPreferenceWhiteList;
+				$defaultOptions = User::getDefaultOptions();
+				$defaultOptionNames = array_keys($defaultOptions);
+
+				return array_reduce(
+					$defaultOptionNames,
+					function($preferences, $option) use ($wgUserPreferenceWhiteList, $defaultOptions) {
+						if (in_array($option, $wgUserPreferenceWhiteList['literals'])) {
+							$preferences[$option] = $defaultOptions[$option];
+						}
+
+						return $preferences;
+					}, []);
```

Comment:
Should we separate this inner block into `User::getDefaultPreferences`? If I understand correctly what's happening we are replacing "get default options" with "get default preferences".